### PR TITLE
Pass ECR Policy with conditional constraint

### LINKED ISSUE #1582: BC_AWS_PUBLIC_1 does not take Conditions into account
**Describe the bug**
BC_AWS_PUBLIC_1 seems to only look for the principal in a ECR Resource Policy. If you want to control access to a repo based on the pricipals AWS Organization ID, you have to do this using a condition instead of a Principal. It seems like checkov does not take this into account.

Example:

``` yaml
  Repository:
    Type: AWS::ECR::Repository
    Properties:
      RepositoryName: !Ref RepoName
      RepositoryPolicyText:
        Version: "2012-10-17"
        Statement:
          - Sid: OrgRead
            Effect: Allow
            Principal: "*"
            Action:
              - ecr:GetAuthorizationToken
              - ecr:BatchCheckLayerAvailability
              - ecr:GetDownloadUrlForLayer
              - ecr:GetRepositoryPolicy
              - ecr:DescribeRepositories
              - ecr:ListImages
              - ecr:DescribeImages
              - ecr:BatchGetImage
              - ecr:DescribeImageScanFindings
            Condition:
              ForAllValues:StringEquals:
                aws:PrincipalOrgID: "o-12345678"
```  

**To Reproduce**
Create the above mentioned resource in cloudformation and scan it with checkov

**Expected behavior**
I expect checkov to look for the usage of a PrincipalOrgID or similar global condition keys and not fail the check when they are used to make the repository non public.

**Desktop (please complete the following information):**
 - OS: macOS 11.5.2
 - Checkov Version 2.0.390

### PR DESCRIPTION
**Pull request by MooreDerek**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Fixes #1582 

Issue 1582 - Allow an AWS ECR Policy to pass with a wildcarded Prinicipal if there is a Principal Org ID constraint conditional statement.

### CODE REVIEW

File: checkov/cloudformation/checks/resource/aws/ECRPolicy.py [hunk 1]
```diff
@@ -31,9 +31,22 @@ def scan_resource_conf(self, conf):
                     for statement_index, statement in enumerate(policy_text['Statement']):
                         if 'Principal' in statement.keys():
                             for principal_index, principal in enumerate(statement['Principal']):
-                                if principal == "*":
+                                if principal == "*" and self.check_for_constrained_condition(statement) == False:
```
Comment (gruebel) [inline checkov/cloudformation/checks/resource/aws/ECRPolicy.py:34]:
```suggestion
                                if principal == "*" and not self.check_for_constrained_condition(statement):
```

File: checkov/terraform/checks/resource/aws/ECRPolicy.py [hunk 2]
```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
                 statement = policy['Statement'][0]
                 if statement['Principal'] and type(statement['Principal']) is str:
                     principal = statement['Principal']
-                    if principal == "*":
+                    if principal == "*"  and self.check_for_constrained_condition(statement) == False:
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/ECRPolicy.py:30]:
```suggestion
                    if principal == "*"  and not self.check_for_constrained_condition(statement):
```

File: checkov/terraform/checks/resource/aws/ECRPolicy.py [hunk 3]
```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
...
     def get_evaluated_keys(self) -> List[str]:
         return ['policy']
 
+    def check_for_constrained_condition(self, statement):
+        """
+        Checks to see if there is a constraint on a a wildcarded principal
+        :param statement: statement from aws_repository_configuration
+        :return: True if there is a constraint
+        """
+        if statement['Condition'] and type(statement['Condition']) is dict:
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/ECRPolicy.py:44]:
```suggestion
        if 'Condition' in statement and isinstance(statement['Condition'], dict):
```

File: checkov/terraform/checks/resource/aws/ECRPolicy.py [hunk 4]
```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
...
+          condition = statement['Condition']
+          if condition['ForAllValues:StringEquals'] and type(condition['ForAllValues:StringEquals']) is dict:
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/ECRPolicy.py:46]: same here, check if the key exists and then check if it is an instance of `dict`

File: checkov/terraform/checks/resource/aws/ECRPolicy.py [hunk 5]
```diff
@@ -27,13 +27,25 @@ def scan_resource_conf(self, conf):
...
-                    if principal == "*":
+                    if principal == "*"  and self.check_for_constrained_condition(statement) == False:
                         self.evaluated_keys = ["policy/Statement/Principal"]
                         return CheckResult.FAILED
         return CheckResult.PASSED
 
     def get_evaluated_keys(self) -> List[str]:
         return ['policy']
 
+    def check_for_constrained_condition(self, statement):
```
Comment (gruebel) [inline checkov/terraform/checks/resource/aws/ECRPolicy.py:38]: also a test for terraform would be great

File: tests/cloudformation/checks/resource/aws/example_ECRPolicy/ECRPolicy-PASSED-2.yaml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        Version: "2012-10-17"
+        Statement: 
+          - 
+            Sid: AllowPushPull
+            Effect: Allow
+            Principal: "*"
+            #Principal: 
+            #   AWS: 
+            #     - "arn:aws:iam::123456789012:user/Bob"
+            #     - "arn:aws:iam::123456789012:user/Alice"
```
Comment (gruebel) [inline tests/cloudformation/checks/resource/aws/example_ECRPolicy/ECRPolicy-PASSED-2.yaml:17]: just remove the comment, not needed anymore 🙂

**Review by gruebel — COMMENTED** [review]
and of course adjust your code to be correctly formatted 👍

File: checkov/terraform/checks/resource/aws/ECRPolicy.py (hunk 5 above)
Comment (MooreDerek) [reply checkov/terraform/checks/resource/aws/ECRPolicy.py:38]: Have added a test for pass and fail on constrained condition

File: checkov/cloudformation/checks/resource/aws/ECRPolicy.py (hunk 1 above)
Comment (MooreDerek) [reply checkov/cloudformation/checks/resource/aws/ECRPolicy.py]: Done

File: checkov/terraform/checks/resource/aws/ECRPolicy.py (hunk 2 above)
Comment (MooreDerek) [reply checkov/terraform/checks/resource/aws/ECRPolicy.py]: Done

File: checkov/terraform/checks/resource/aws/ECRPolicy.py (hunk 3 above)
Comment (MooreDerek) [reply checkov/terraform/checks/resource/aws/ECRPolicy.py]: Done

File: checkov/terraform/checks/resource/aws/ECRPolicy.py (hunk 4 above)
Comment (MooreDerek) [reply checkov/terraform/checks/resource/aws/ECRPolicy.py]: Done

File: checkov/terraform/checks/resource/aws/ECRPolicy.py (hunk 5 above)
Comment (MooreDerek) [reply checkov/terraform/checks/resource/aws/ECRPolicy.py:38]: Has been added :)

File: tests/cloudformation/checks/resource/aws/example_ECRPolicy/ECRPolicy-PASSED-2.yaml (hunk 6 above)
Comment (MooreDerek) [reply tests/cloudformation/checks/resource/aws/example_ECRPolicy/ECRPolicy-PASSED-2.yaml]: Done

**Review by gruebel — APPROVED** [review]
thanks for the contribution 🥇

**Pull request merged**